src/api/local-storage-provider.model.ts

```diff
@@ -8,74 +8,74 @@ import {
   ScheduleRevision,
   ThunkFunction,
   ScheduleKey,
-  ScheduleRecord,
+  RevisionType,
 } from "./persistance-store.model";
-/*eslint-disable @typescript-eslint/no-unused-vars*/
+import PouchDB from "pouchdb-browser";
 
-export class LocalStorageProvider implements PersistanceStoreProvider {
-  private sleep(ms: number): Promise<void> {
-    return new Promise((resolve) => setTimeout(resolve, ms));
+/*eslint-disable @typescript-eslint/camelcase */
+export class LocalStorageProvider extends PersistanceStoreProvider {
+  private storage: PouchDB.Database<ScheduleRevision>;
+
+  constructor() {
+    super();
+    this.storage = new PouchDB("nurse-scheduling");
   }
-  saveScheduleRevision(revision: ScheduleRevision): ThunkFunction<ScheduleDataModel> {
+
+  saveScheduleRevision(
+    type: RevisionType,
+    schedule: ScheduleDataModel
+  ): ThunkFunction<ScheduleDataModel> {
     return async (dispatch, getState): Promise<void> => {
-      await this.sleep(100);
-      const revisions = JSON.parse(localStorage.getItem("revisions") || "[]");
-      localStorage.setItem("currentSchedule", JSON.stringify([...revisions, revision]));
+      const id = this.getScheduleId(schedule);
+      let revision = "";
+      try {
+        const document = await this.storage.get(id);
+        revision = document._rev;
+      } catch {}
+      this.storage.put({
+        _rev: revision,
+        _id: id,
+        data: schedule,
+        revisionType: type,
+      });
       dispatch({
         type: ScheduleDataActionType.ADD_NEW,
-        payload: revision.data,
+        payload: schedule,
       });
     };
   }
+
   getScheduleRevision(filter: RevisionFilter): ThunkFunction<ScheduleDataModel> {
     return async (dispatch, getState): Promise<void> => {
-      await this.sleep(100);
-      const revisions: ScheduleRecord[] = JSON.parse(localStorage.getItem("revisions") || "[]");
-      const result = revisions.find(
-        (r) =>
-          r.validityPeriod.month === filter.validityPeriod.month &&
-          r.validityPeriod.year === filter.validityPeriod.year
-      );
-      if (!result) {
+      const revisions = await this.storage.allDocs({ include_docs: true });
```

Comment:
Wiem, że dziwnie wygląda: pobieranie wszystkich dokumentów z bazy, ale to jest polecana w dokumentacji metoda do napisania zapytań.
Ale teraz zacząłem się zastanawiać i pewnie coś źle rozumiem, więc jeszcze o tym doczytam
https://pouchdb.com/guides/bulk-operations.html#please-use-alldocs